src/Chapter1.hs

```diff
@@ -553,8 +542,11 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | (y < x && x < z) || (z < x && x < y) = x
+    | (x < y && y < z) || (z < y && y < x) = y
+    | (x < z && z < y) || (y < z && z < x) = z
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = ...`, to tell the compiler, that your pattern matching is exhaustive 🙂
And as you correctly notices, you have already covered all cases, so actually, you don't need the last check, you know that it would be `z`:

```suggestion
    | otherwise = z
```